Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -0,0 +1,320 @@
+import Sentry
+import XCTest
+
+#if os(iOS) || os(macOS) || targetEnvironment(macCatalyst)
+class SentryProfilerSwiftTests: XCTestCase {
+    private static let dsnAsString = TestConstants.dsnAsString(username: "SentryProfilerSwiftTests")
+
+    private class Fixture {
+        lazy var options: Options = {
+            let options = Options()
+            options.dsn = SentryProfilerSwiftTests.dsnAsString
+            return options
+        }()
+        lazy var client: TestClient! = TestClient(options: options)
+        lazy var hub: SentryHub = {
+            let hub = SentryHub(client: client, andScope: scope)
+            hub.bindClient(client)
+            return hub
+        }()
+        let scope = Scope()
+        let message = "some message"
+        let transactionName = "Some Transaction"
+        let transactionOperation = "Some Operation"
+    }
+
+    private var fixture: Fixture!
+
+    override func setUp() {
+        super.setUp()
+        fixture = Fixture()
+        SentryTracer.resetAppStartMeasurementRead()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        clearTestState()
+        SentryTracer.resetAppStartMeasurementRead()
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+        SentryFramesTracker.sharedInstance().resetFrames()
+        SentryFramesTracker.sharedInstance().stop()
+#endif
+    }
+
+    // This test is only available on newer APIs because of the availability of `DispatchQueue.activate`
+    @available(tvOS 10.0, *)
+    @available(OSX 10.12, *)
+    @available(iOS 10.0, *)
+    func testConcurrentProfilingTransactions() {
+        let options = fixture.options
+        options.profilesSampleRate = 1.0
+        options.tracesSampleRate = 1.0
+
+        let queue = DispatchQueue(label: "SentryProfilerSwiftTests", attributes: [.concurrent, .initiallyInactive])
+        let group = DispatchGroup()
+
+        let numberOfTransactions = 10
+        for _ in 0 ..< numberOfTransactions {
+            group.enter()
+            let exp = expectation(description: "finished span")
+            let span = fixture.hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
+
+            // Some busy work to try and get it to show up in the profile.
+            let str = "a"
+            var concatStr = ""
+            for _ in 0..<100_000 {
+                concatStr = concatStr.appending(str)
+            }
+
+            queue.asyncAfter(deadline: .now() + 2) {
```

Comment:
`l`: 2 seconds seems a bit long for me. Can we reduce this a bit to make the test faster?

Reply:
I've often wondered what would happen if I were to check this box: 
<img width="231" alt="image" src="https://user-images.githubusercontent.com/3241469/190564536-940ee4bd-3bf1-44d1-97ec-3aeedcb97227.png">
Let's see what happens, this might save much more time than changing this delay. 

Also, I had to add two tests for timeouts, which run much longer. Maybe I can try using `TestCurrentDateProvider` for those.

Reply:
Parallel doesn't work, most likely, due to a lot of shared global state. Already pointed that out here https://github.com/getsentry/sentry-cocoa/pull/2175#issuecomment-1248941657

Reply:
I don't understand why we need this delay. If you want to travel 2 seconds in time, please use the `TestCurrentDateProvider` as pointed out by you. If you want to have 10 concurrent transactions, I would rather put all the code in the loop into `asyncAfter` and only use the test expectation to wait until all transactions are finished. Something like this should work
```Swift
let queue = DispatchQueue(label: "SentryProfilerSwiftTests", attributes: [.concurrent, .initiallyInactive])
let numberOfTransactions = 10
let exp = expectation(description: "finished transaction")
exp.expectedFulfillmentCount = numberOfTransactions
for _ in 0 ..< numberOfTransactions {
    queue.async {
        let transaction = self.fixture.hub.startTransaction(name: self.fixture.transactionName, operation: self.fixture.transactionOperation)
        
        // Some busy work to try and get it to show up in the profile.
        let str = "a"
        var concatStr = ""
        for _ in 0..<100_000 {
            concatStr = concatStr.appending(str)
        }
        
        transaction.finish()
        exp.fulfill()
    }
}

queue.activate()
waitForExpectations(timeout: 1.0)
```